Makefile.am

```diff
@@ -23,3 +23,7 @@ ACLOCAL_AMFLAGS = -Im4/autotools
 SUBDIRS = m4 include lib src doc man
 dist_doc_DATA = README.md
 dist_noinst_DATA = tlog.spec
+# Remove *.gcno files from all directories.
+clean-local:
+	-rm -rf *.gcno
```

Comment:
This is not how shell and rm works. It won't do what I think you tried to do here. You need to understand how shell does expansion. See https://www.gnu.org/software/bash/manual/html_node/Shell-Expansions.html and ping me with questions if needed.

Then, this Makefile.am is only responsible for the root directory directly. Each subdirectory has its own Makefile.am's to care for it. You need to add removal of .gcno files to Makefile.am of each directory, which has .gcno files produced. I.e. each directory which has .c files.